mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go

```diff
@@ -141,6 +142,45 @@ func TestAccLoggingProjectSink_updatePreservesUniqueWriter(t *testing.T) {
 		},
 	})
 }
+*/
+
+func TestAccLoggingProjectSink_updatePreservesCustomWriter(t *testing.T) {
+	t.Parallel()
+
+	sinkName := "tf-test-sink-" + acctest.RandString(t, 10)
+	account := "tf-test-sink-sa" + acctest.RandString(t, 10)
+
+	org := envvar.GetTestOrgFromEnv(t)
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+	project := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckLoggingProjectSinkDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLoggingProjectSink_customWriter(org, billingId, project, sinkName, account),
+			},
+			{
+				ResourceName:      "google_logging_project_sink.custom_writer",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// Logging sink create API doesn't return this field in response
```

Comment:
qq/clarification: It's returned through `writerIdentity` on https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.sinks but that could also be the server-determined result so we can't use that to detect drift, right?

Reply:
Correct. it's server determined field that could be either uniqueWriterIdentity or customWriterIdentity depending on how user configures the sink. However, I ran into testing error where the test cases expect the customWriterIdentity field in the import (or local) state. Thus, using the ImportStateVerifyIgnore.

Reply:
We definitely have to use ImportStateVerifyIgnore for custom_writer_identity, since it's write-only. However, since writer_identity is already exposed as an output-only field, I believe you can use [TestCheckFuncs](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep#check-functions) to 
verify that the value of writer_identity matches the expected custom_writer_identity.

Reply:
Added the TestCheckFuncs for both create and patch tests.